portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,54 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns some example content. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> comments = new ArrayList<>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Ryan!</h1>");
+    Query query = new Query("Comment").addSort("message", SortDirection.DESCENDING);
```

Comment:
It's good practice to keep most Strings and numeric values as constants at the top of the class, or in their own file if they need to be shared across files. The two main reasons:
1. They get to have names, which can serve as a form of documentation in order to improve readability. If the meaning of the value requires some extra explanation, declaring it in one spot gives you a place to put a descriptive comment. Here's an example of naming + comment to add documentation: [DESIRED_LOAD_FACTOR](https://github.com/google/guava/blob/794a10a4495fd7cd554eb0a9e480708d0277aced/guava/src/com/google/common/collect/ImmutableSet.java#L630)
2. It protects against bugs caused by typos when trying to use the same value in multiple places, e.g. I'll add a parameter `entity=foo` to a URL, then wondering why `getParam("enity");` isn't returning anything. At least if I made a typo when referring to the name of the constant, the compiler will let me know.

Could you define and use most of the values used in this file as constants?  The standard format is basically like the `DESIRED_LOAD_FACTOR`, except that file has a bunch of fancy-pants abstract class stuff, so for a more normal file like this one, constants should be put at the top, i.e. right after `public class DataServlet extends HttpServlet {`.

I think names of entity properties, the entity kind, and request parameter keys should all be constants. For the content type values, I could see the case for making them constants or leaving them as hardcoded inline Strings. For the HTML tags, I think it's most clear to leave as-is rather than make them into constants.

Edit: I see that some of these are used in the deletion servlet, so for those ones you can make a shared location, e.g. Constants.java which contains a Constants class to hold anything that should be shared.